### ISSUE TITLE
Multidimensional structure tensor

### ISSUE THREAD

**Issue by 0x00b1** [scikit-image/scikit-image#2346]
[skimage.feature.structure_tensor](http://scikit-image.org/docs/dev/api/skimage.feature.html#structure-tensor) requires a two-dimensional array.

``` Python
In [1]: import numpy

In [2]: import skimage.feature

In [3]: shape = (5, 5, 5)

In [4]: cube = numpy.zeros(shape)

In [5]: cube[2, 2, 2] = 1

In [6]: cube
Out[6]: 
array([[[ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.]],

       [[ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.]],

       [[ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  1.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.]],

       [[ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.]],

       [[ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.],
        [ 0.,  0.,  0.,  0.,  0.]]])

In [7]: skimage.feature.structure_tensor(cube)
ValueError: The parameter `image` must be a 2-dimensional array
```

It’d be useful to compute the structure tensor of a multidimensional array (e.g. a three-dimensional volume) because it’s a common feature for texture representation and regularly used to solve multidimensional imaging problems (e.g. image registration).

**Comment by 0x00b1** [comment]
If anybody else is interested, I started to experiment by adapting skimage.feature.structure_tensor:

https://gist.github.com/0x00b1/a96ebe7b6dded58c88e071b9bd7e6a01

**Comment by soupault** [comment]
During the implementation it might be better to use r/c convention for array manipulating (see https://github.com/scikit-image/scikit-image/issues/2301).

**Comment by grlee77** [comment]
closed by #5002
